### PR TITLE
Stats error when all the calls fail

### LINKED ISSUE #9038: Polish errors in refreshed Stats
* When all the blocks fail, we want to show a full screen error with a retry button
* We never want to hide a block with missing data - there should always be a placeholder with the "No data yet" message
* We want to stop loading of the DWMY blocks if there are no dates in the Overview block

### PR DESCRIPTION
**Pull request by planarvoid**
Fixes #9038 

I tried several approaches to this and I think this is the best we can do. Some of the code could probably be extracted but after the change it's consistent across the blocks and works better than before. It is too long PR because we have so many usecases and the changes are the same in all of them.

* Show a fullscreen error when loading of all blocks fails. The block contains a "Retry" button which refreshes the data. 
* Show all the blocks even if they miss data - for example Latest post summary was disappearing before
* Stop loading of the DWMY blocks if the Overview usecase fails or has no data.
--
To test:
* Turn on the airplane mode
* Go to stats
* You see the empty screen with the "Retry" button
* Turn off the airplane mode
* Click "Retry"
* The screen gets reloaded with the fresh data

To test 2:
* Open stats
* Turn on the airplane mode
* Pull to refresh
* The cached data doesn't disappear

To test 3:
* Try variations of tabs and airplane mode state - selecting date, turning AM on and off, refreshing the screen.

### CODE REVIEW

**Review by 0nko — COMMENTED** [review]
Nice PR. Here's a couple of observations from testing:

- We could specifically handle the offline mode and show a more relevant message than "Data not loaded". You could use `NetworkUtilsWrapper` for this in the ViewModels.
- When you tap the Retry button flight mode, the screen blinks momentarily, then shows the error again. I think for the offline case, we could always first verify the network connection before attempting to do anything else.
- When you pull to refresh while offline, I can see the cached data but there is no indication that something failed. We could pop snackbar message, telling the user the device's offline or or something else failed.
- When you load the data first, then switch on the flight mode and then attempt to change the Followers tab to Email, the block just disappears.
- When you load the data first, then switch on the flight mode and then tap on the Overview graph column, the cached data disappears and the error message is shown. Maybe we could just keep showing the cached data and just pop an error message.

**Comment by planarvoid** [comment]
@0nko this should be ready for another round of testing. It was really tricky to implement it right but at the moment when everything fails on the Insights tab and you already have the cached data, we show the cached data. On the DWMY tab when you are switching the selected date and you lose internet connection, the `Overview` block will still be visible and showing the selected date but all the other blocks will show the error. I think this makes the most sense.
Everything you suggested should be now fixed too.
Thanks for the testing 👍

**Comment by 0nko** [comment]
It works much better now. The only thing left I think is show some offline error message when you try to pull-to-refresh in DMWY (currently nothing happens).

**Comment by planarvoid** [comment]
@0nko there should be a snackbar now when you try to PTR in DWMY and you're offline.

**Comment by 0nko** [comment]
Thanks for the changes! I just noticed one more thing - the snackbar is now displayed twice when you open stats while offline.

**Comment by planarvoid** [comment]
@0nko thanks for checking! This should be fixed now. I moved the snackbar logic to the parent `StatsViewModel` instead of having it in the list fragments. This way we only show the snackbar once even when all the independent fragments fails.

File: WordPress/src/main/java/org/wordpress/android/ui/stats/refresh/StatsFragment.kt [hunk 1]
```diff
@@ -147,6 +147,23 @@ class StatsFragment : DaggerFragment() {
...
+                val snackbar = Snackbar.make(parent, message.message, Snackbar.LENGTH_LONG)
+                val snackbarText = snackbar.view.findViewById<TextView>(android.support.design.R.id.snackbar_text)
+                snackbarText.maxLines = 2
+                snackbar.show()
+                true
+            } else {
+                false
+            }
+        }
+         */
```
Comment (0nko) [inline WordPress/src/main/java/org/wordpress/android/ui/stats/refresh/StatsFragment.kt]: We can probably remove this.

**Review by 0nko — APPROVED** [review]
Besides a minor comment, everything looks good. Thanks! 👍

**Pull request merged**